legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/test/java/org/finos/legend/engine/language/pure/compiler/test/TestDataSpaceCompilationFromGrammar.java

```diff
@@ -988,6 +988,143 @@ public void testDataSpaceWithTemplateExecutable()
+                "      executionContextKey: 'Context 1';\n" +
+                "    },\n" +
+                "    {\n" +
+                "      id: 2;\n" +
+                "      title: 'Template 2';\n" +
+                "      query: model::templateFunc():TabularDataSet[1];\n" +
+                "    },\n" +
+                "    {\n" +
+                "      id: 3;\n" +
+                "      title: 'Template 3';\n" +
+                "      description: 'an example of a template query';\n" +
+                "      query: model::templateFunc():TabularDataSet[1];\n" +
+                "    }\n" +
+                "  ];\n" +
+                "}\n");
+
+        test(model +
+                "###DataSpace\n" +
+                "DataSpace model::dataSpace" +
+                "{\n" +
+                "  executionContexts:\n" +
+                "  [\n" +
+                "    {\n" +
+                "      name: 'Context 1';\n" +
+                "      description: 'some information about the context';\n" +
+                "      mapping: model::dummyMapping;\n" +
+                "      defaultRuntime: model::dummyRuntime;\n" +
+                "    }\n" +
+                "  ];\n" +
+                "  defaultExecutionContext: 'Context 1';\n" +
+                "  elements: [model::element, model, model::sub];\n" +
+                "  executables:\n" +
+                "  [\n" +
+                "    {\n" +
+                "      id: 1;\n" +
+                "      title: 'Template 1';\n" +
+                "      query: model::templateFunc1():TabularDataSet[1];\n" +
+                "      executionContextKey: 'Context 1';\n" +
+                "    }\n" +
+                "  ];\n" +
+                "}\n", " at [44:1-65:1]: Error in 'model::dataSpace': The mapping utilized in the function within the curated template query does not align with the mapping applied in the execution context.");
+
+        test(model +
+                "###DataSpace\n" +
+                "DataSpace model::dataSpace" +
+                "{\n" +
+                "  executionContexts:\n" +
+                "  [\n" +
+                "    {\n" +
+                "      name: 'Context 1';\n" +
+                "      description: 'some information about the context';\n" +
+                "      mapping: model::dummyMapping;\n" +
+                "      defaultRuntime: model::dummyRuntime;\n" +
+                "    }\n" +
+                "  ];\n" +
+                "  defaultExecutionContext: 'Context 1';\n" +
+                "  elements: [model::element, model, model::sub];\n" +
+                "  executables:\n" +
+                "  [\n" +
+                "    {\n" +
+                "      id: 1;\n" +
+                "      title: 'Template 1';\n" +
+                "      query: model::templateFunc2():TabularDataSet[1];\n" +
+                "      executionContextKey: 'Context 1';\n" +
```

Comment:
Lets think about what we want to support here. 
Should we support standalone TDS functions without the `->from()` with mapping/runtime and just apply the from from the executionContext provided ? 
If they do provide `->from()` then your check below makes sense but then they should not provide an executoinContextkey and we should throw if they do ?